pynestml/symbols/type_symbol.py

```diff
@@ -191,7 +191,10 @@ def is_castable_to(self, _other_type):
         :return: True if castable, otherwise False
         :rtype: bool
         """
-        pass
+        from pynestml.symbols.template_type_symbol import TemplateTypeSymbol
+        if isinstance(_other_type, TemplateTypeSymbol):
+            return True
```

Comment:
why not call the method from ``_other_type``? They are not necessarily castable but the relationship is commutative